Home page, structure and formatting improvements (#54)
nhsbsa-data-analytics/nhsbsa-analytical-code-assurance-playbook: guidance/07-open-source/index.qmd

---
title: "Open Source"
description: "Open source code can be viewed, downloaded, used and changed by anyone."
---

## At a glance

> Open source code can be viewed, downloaded, used and changed by anyone. 

Reviewer: should we make clear here that changed by anyone doesn't mean making direct changes to the codebase?

Author: ```suggestion
Open source code can be viewed, downloaded and used by anyone. 
```

Author: Suggestion above

This guide outlines the benefits of publishing analytical code as open source and how to do it within the NHSBSA.

## What is open source?

Open source is a way of developing and distributing software. 

> The code can be viewed, downloaded, used and changed by anyone.

Reviewer: same as previous comment

Author: ```suggestion
The code can be viewed, downloaded and used by anyone. Anyone can contribute to open source code (pending approval by the code owner via Pull Requests) which promotes collaboration across teams and organisations.
```

Author: Suggestion above

Open source code is often hosted on platforms like GitHub, where it can be shared and collaborated on by a community of developers.
Open source code is typically licensed under an open source licence, which specifies how others can use, modify and share the code.

## Why publish analytical code as open source?

Open source code:

* Allows others to see how analyses are performed, increasing transparency and trust in results.
* Shows the workings behind published analyses (reports, data, dashboards, etc), making it easier to verify and reproduce results
* Enables others to reuse code which increases value.
* Open review and feedback help identify issues and improve code.

> * Encourages collaboration and knowledge sharing across teams and organisations.

Reviewer: ```suggestion
* allows others to see how analyses are performed, increasing transparency and trust in results
* shows the workings behind published analyses (reports, data, dashboards, etc), making it easier to verify and reproduce results
* enables others to reuse code which increases value
* open review and feedback help identify issues and improve code
* encourages collaboration and knowledge sharing across teams and organisations
```
remove capitalisation and punctuation.

also, the first 2 bullets feel like the same thing said in 2 different ways? doesn't necessarily need to change but worth trying to see if we can make a clearer distinction

> To publish analytical code:

Reviewer: i can't make suggestions to the below block as a whole for some reason so have done point by point. amendments to remove AI formatting types

Author: Have applied.

> 1. **Include a full LICENCE file:** Add a file named `LICENCE` (British English spelling) with the complete licence text in your repository, not just a statement in the `README` (see below to learn more about the licences we use at the NHSBSA).

Reviewer: ```suggestion
1. Include a file named `LICENCE` (British English spelling) with the complete licence text in your repository, not just a statement in the `README`.
```

> 1. **Document your code:** Provide clear instructions for use, dependencies, and purpose. Ensure the code itself is written to be easy to understand and maintain, and include a `CONTRIBUTE` file to guide others who may want to contribute.

Reviewer: ```suggestion
2. Document your code and provide clear instructions for use, dependencies, and purpose. Ensure the code itself is written to be easy to understand and maintain, and include a `CONTRIBUTE` file to guide others who may want to contribute.
```

> 1. **Host on GitHub:** Host your code on the [NHSBSA Data Analytics GitHub][nhsbsa-github].

Reviewer: ```suggestion
3. Host your code on the [NHSBSA Data Analytics GitHub][nhsbsa-github].
```

> 1. **Follow NHSBSA guidance:** Adhere to [NHSBSA policies][nhsbsa-open-source] and approval processes for open source coding (only visible to internal NHSBSA colleagues).

Reviewer: ```suggestion
4. Follow [NHSBSA policies][nhsbsa-open-source] and approval processes for open source coding.
```

> 1. **Avoid storing secret keys or credentials in source code:** Use secret management systems and keep credentials out of repositories. Make sure you check the history as well as the current version of the codebase! See [GOV.UK guidance][govuk-secret-guidance].

Reviewer: ```suggestion
5. Use secret management systems and keep credentials out of repositories. Make sure you check the history as well as the current version of the codebase! See [GOV.UK guidance][govuk-secret-guidance].
```
